src/test/pybind/test_rbd.py

```diff
@@ -1082,7 +1082,7 @@ def test_set_snap_recreated(self):
         eq(read, data)
         self.image.set_snap('snap1')
         self.image.remove_snap('snap1')
-        self.image.create_snap('snap1')
+        assert_raises(ImageNotFound, self.image.create_snap, 'snap1')
```

Comment:
We can't change the existing behaviour of the API. We should be able to create a snapshot here.

Reply:
@dillaman  Yeah, I was thinking we could not propagate back the -ENOENT error in [InvalidateRequest](https://github.com/ceph/ceph/pull/34059/files#diff-1e91307bc8cf81a68ca4c24258347e5dL34) (but it seems you don't like it https://github.com/ceph/ceph/pull/34059#discussion_r399378444), so ...
I still prefer to not propagate back the -ENOENT error in InvalidateRequest, what do you think? thank you.

Reply:
I thought that was covered by my comment here [1]. What's the source of this `-ENOENT`? 


[1] https://github.com/ceph/ceph/pull/34059#discussion_r399380790

Reply:
@dillaman  Removing an in-used (via `image.set_snap('snap1')`) snapshot will invalidated the image's object-map, and `image_ctx.get_flags(m_snap_id, &snap_flags)` [1] will return `-ENOENT` [2] when recreating a snapshot.

I have another way to resolve it as following (if you don't like this repushd fix [3], any help or suggestions would be appreciated.) Thank you inadvance.
```
--- a/src/librbd/object_map/Request.h
+++ b/src/librbd/object_map/Request.h
@@ -30,7 +30,8 @@ protected:

   bool should_complete(int r) override;
   int filter_return_code(int r) const override {
-    if (m_state == STATE_REQUEST) {
+    if (m_state == STATE_REQUEST ||
+        ((m_state == STATE_INVALIDATE) && r == -ENOENT)) {
       // never propagate an error back to the caller
       return 0;
     }
```

[1] [InvalidateRequest.cc#L30](https://github.com/ceph/ceph/blob/master/src/librbd/object_map/InvalidateRequest.cc#L30)
[2] [In the case where the snapshot object map was never created](https://github.com/ceph/ceph/pull/34059#discussion_r399380790)
[3] https://github.com/ceph/ceph/pull/34059/commits/374be22d267aaa57a3d081585c0fccef497651a5#diff-616fb4f060117fd81715aa3132a52454R58